flask_inertia/inertia.py

```diff
@@ -61,13 +66,29 @@ def init_app(self, app: Flask):
         """
         self.app = app
         self._shared_data = {}
-        if not hasattr(app, "extensions"):
-            app.extensions = {}
-        app.extensions["inertia"] = self
+        if isinstance(app, Flask):
+            self._init_extension(app)
+        elif isinstance(app, Blueprint):
+            blueprint = app
+            # Register the extension once the blueprint is registered
+            blueprint.record_once(self.register_blueprint)
         app.context_processor(self.context_processor)
         app.before_request(self.process_incoming_inertia_requests)
         app.after_request(self.update_redirect)
 
+    def register_blueprint(self, state: BlueprintSetupState):
+        # TODO: support multiple instances of Inertia, each on different
+        #  blueprints? Possibly by using blueprint's name as part of the
+        #  key in app.extensions. e.g:
+        #  app.extensions[f"inertia.{state.blueprint.name}"] = self
+        self._init_extension(state.app)
```

Comment:
What are your thoughts on this? Not sure if there is a valid use-case for multiple `Inertia` instances on multiple blueprints.

The way I have it now, it would silently break in that case. Each blueprint registration would clobber the previous.

Reply:
Or if you can find a way to store a reference to the `Inertia` instance on the blueprint, that would be better. I didn't see any in the flask docs though. There is no `extensions` on Blueprints.

Reply:
The only reason I would see using this extension on a Blueprint instead of on the current app is a migration plan since you can use the extension alongside a normal Flask app. You don't need to support multiple extension instances.